atlas-json-parent/atlas-json-core/src/main/java/io/atlasmap/json/inspect/SchemaInspector.java

```diff
@@ -0,0 +1,124 @@
+    private JsonDocument jsonDocument = AtlasJsonModelFactory.createJsonDocument();
+    private static final Logger logger = LoggerFactory.getLogger(SchemaInspector.class);
+
+    public SchemaInspector() {
+    }
+
+    public JsonDocument getJsonDocument() {
+        return jsonDocument;
+    }
+
+    public void inspect(String schema) throws JsonInspectionException {
+        if (schema == null || schema.isEmpty()) {
+            throw new IllegalArgumentException("JSON schema cannot be null");
+        }
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(schema);
+            
+            JsonNode header = rootNode.get("$schema");
+            if (header == null || !header.asText().startsWith("http://json-schema.org/")) {
+                throw new JsonInspectionException(String.format("The $schema property not found or invalid: '%s'", header.asText()));
+            }
+            JsonNode type = rootNode.get("type");
+            if (type == null || !type.asText().equals("object")) {
+                throw new JsonInspectionException(String.format("The property type '%s' is not supported for root node", type));
+            }
+
+            Iterator<Entry<String, JsonNode>> topFields = rootNode.get("properties").fields();
+            while (topFields.hasNext()) {
+                Entry<String, JsonNode> entry = topFields.next();
+                if (!entry.getValue().isObject()) {
+                    logger.warn("Ignoring non-object field '{}'", entry);
+                    continue;
+                }
+                logger.trace("--> Adding a field '{}' on a root node with value '{}'", entry.getKey(), entry.getValue());
+                JsonComplexType target = createJsonComplexType(entry.getKey(), entry.getValue(), null);
+                jsonDocument.getFields().getField().add(target);
+            }
+        } catch (Exception e) {
+            throw new JsonInspectionException(e);
+        }
+    }
+
+    private JsonComplexType createJsonComplexType(String name, JsonNode value, JsonComplexType parent) throws JsonInspectionException {
+        if (value.get("$ref") != null) {
+            logger.warn("'$ref' is not yet supported for JSON Schema, will be ignored: '{}", value);
+        }
+        JsonNode fieldType = value.get("type");
+        if (fieldType == null || fieldType.asText() == null) {
+            throw new JsonInspectionException(String.format("The type is not defined for node '%s'", name));
+        }
+        
+        JsonComplexType answer = new JsonComplexType();
+        answer.setJsonFields(new JsonFields());
+        answer.setName(name);
+        answer.setStatus(FieldStatus.SUPPORTED);
+        answer.setPath((parent != null ? parent.getPath() : "").concat("/").concat(name));
+        
+        switch (fieldType.asText()) {
+        case "array":
+            answer = createJsonComplexType(name, value.get("items"), parent);
```

Comment:
`value.get("items")` might be null, but this method assumes a not-null `value` element (see above). might be good to either check here for null, or above assert that `value` is not null.